server/src/main/java/org/opensearch/search/pipeline/SearchPipelineService.java

```diff
@@ -390,13 +392,22 @@ public PipelinedRequest resolvePipeline(SearchRequest searchRequest) {
             if (searchRequest.pipeline() != null) {
                 // Named pipeline specified for the request
                 pipelineId = searchRequest.pipeline();
-            } else if (state != null && searchRequest.indices() != null && searchRequest.indices().length == 1) {
+            } else if (state != null && searchRequest.indices() != null) {
                 // Check for index default pipeline
-                IndexMetadata indexMetadata = state.metadata().index(searchRequest.indices()[0]);
-                if (indexMetadata != null) {
-                    Settings indexSettings = indexMetadata.getSettings();
-                    if (IndexSettings.DEFAULT_SEARCH_PIPELINE.exists(indexSettings)) {
-                        pipelineId = IndexSettings.DEFAULT_SEARCH_PIPELINE.get(indexSettings);
+                Index[] concreteIndices = indexNameExpressionResolver.concreteIndices(state, searchRequest);
```

Comment:
If `searchRequest.indices()` is empty here, it goes to this line of code and applies to _all_ indices:
```java
    Index[] concreteIndices(Context context, String... indexExpressions) {
        if (indexExpressions == null || indexExpressions.length == 0) {
            indexExpressions = new String[] { Metadata.ALL };
        }
```
I don't think that's what we want.

And we should have a unit test on this edge case.

Reply:
Added the check and a unit test

Reply:
As you've seen in your jenkins test failure, this is throwing an exception if an index name doesn't resolve, so this whole section needs to be in a try/catch block (ignoring the unmatched index and just not applying a default pipeline)

Reply:
Yes, added a try/catch block to handle the exception. Also, a added a UT for invalid index